### PR TITLE
final set of proofreading changes

### PR DESCRIPTION
**Pull request by cgreene**

### CODE REVIEW

File: sections/06_discussion.md [hunk 1]
```diff
@@ -462,34 +400,35 @@ efficiency have come from computation with graphics processing units (GPUs)
...
+biological datasets and increasingly complex neural networks. Much has been
 done to minimize the memory requirements of neural networks [@tag:CudNN
 @tag:Caruana2014_need @tag:Gupta2015_prec @tag:Bengio2015_prec
 @tag:Sa2015_buckwild @tag:Chen2015_hashing @tag:Hubara2016_qnn], but there is
 also growing interest in specialized hardware, such as field-programmable gate
 arrays (FPGAs) [@tag:Edwards2015_growing_pains @tag:Lacey2016_dl_fpga] and
-application-specific integrated circuits (ASICs) [@arxiv:1704.04760]. Specialized hardware promises
+application-specific integrated circuits (ASICs) [@arxiv:1704.04760].
+Less software
+available for such highly specialized hardware [@tag:Lacey2016_dl_fpga].
```
Comment (agitter) [inline sections/06_discussion.md]: "is available"

File: sections/06_discussion.md [hunk 2]
```diff
@@ -584,49 +523,52 @@ Transcription Factor Binding Site Prediction Challenge [@tag:Dream_tf_binding],
...
-Neural Networks), offers not only code and a model zoo but also a detailed
+Neural Networks) offers not only code and a model zoo but also a detailed
 tutorial and partner package for simulating training data.  These resources,
 especially the ability to simulate datasets that are sufficiently complex to
 demonstrate the challenges of training neural networks but small enough to train
-quickly on a CPU, are important for (human) training and attracting machine
-learning researchers to problems in genomics and healthcare.  We have even
+quickly on a CPU, are important for training students and attracting machine
+learning researchers to problems in genomics and healthcare.
+`TODO: Cite syllabus or this last sentence should probably go. Unclear what it
```
Comment (agitter) [inline sections/06_discussion.md]: Please cut the last sentence about DragoNN in the course

File: sections/07_conclusions.md [hunk 3]
```diff
@@ -9,19 +9,18 @@ problem being addressed, we conclude that deep learning has not *yet* realized
...
 As an analogy, consider recent progress in conversational speech recognition.
-Since 2009 there have been drastic performance improvements, with error rates
+Since 2009 there have been drastic performance improvements with error rates
 dropping from more than 20% to less than 6% [@tag:Speech_recognition] and
 finally approaching or exceeding human performance in the past year
 [@arxiv:1610.05256 @arxiv:1703.02136]. The phenomenal improvements on benchmark
 datasets are undeniable, but greatly reducing the error rate on these benchmarks did not
 fundamentally transform the domain.  Widespread adoption of conversational
-speech technologies will require not only improvements over baseline methods but
-truly solving the problem, in this case exceeding human-level performance, as
```
Comment (agitter) [inline sections/07_conclusions.md]: This phrasing was awkward, but I wrote it with such complexity to emphasize "solving the problem".  In speech recognition, human performance may be the goal line for solving the problem.  For many tasks we review, human performance isn't relevant though.  Is there a way to convey that more concisely?

File: sections/07_conclusions.md [hunk 4]
```diff
@@ -52,32 +51,28 @@ Conversely, the most challenging tasks may be those in which predictions are
...
 @arxiv:1412.6572], a reminder that these algorithms do not understand the
-semantics of the objects presented. Despite progress in addressing some of these
-limitations [@arxiv:1611.03814 @arxiv:1704.01155], we expect human experts will retain
-their critical roles in the laboratory and clinic
-for the foreseeable future. In some cases, cooperation
-between human experts and deep learning algorithms can achieve better
-performance than either individually [@arxiv:1606.05718]. Especially for sample
-and patient classification tasks, we expect deep learning methods to complement
-and assist biomedical researchers rather than compete with or even replace them.
+semantics of the objects presented. Work in this area is continuing
```
Comment (agitter) [inline sections/07_conclusions.md]: I think the phrase "work in this area" should make it clear that the ongoing work is to guard against attacks and adversarial examples.  Maybe the second half of the sentence is clear enough though?

File: sections/07_conclusions.md [hunk 5]
```diff
@@ -52,32 +51,28 @@ Conversely, the most challenging tasks may be those in which predictions are
...
+be impossible to guarantee that a model is not susceptible to adversarial
+examples. Cooperation
+between human experts and deep learning algorithms addresses many of these challenges and can achieve better
+performance than either individually [@arxiv:1606.05718]. For sample
+and patient classification tasks, we expect deep learning methods to augment
+clinicians and biomedical researchers.
 
-Even if deep learning in biology and healthcare is not yet transformative today,
-we are extremely optimistic about its future. Given how rapidly deep learning is
+We are extremely optimistic about the future of deep learning in biology and medicine. Given how rapidly deep learning is
```
Comment (agitter) [inline sections/07_conclusions.md]: I'll save it for a later issue or pull request, but some of my extreme enthusiasm has dampened a little after re-reading and reflecting upon the entire review.  Let's merge this as-is and discuss separately.

File: sections/06_discussion.md [hunk 6]
```diff
@@ -123,93 +119,41 @@ quantifying the change in the output. Umarov et al.
...
+fewer iterations but
 requires the perturbation to have a differentiable form.
 
-##### Backpropagation-based approaches
-
-A second strategy for addressing the computational inefficiency of
-perturbation-based approaches is to propagate an important signal from a target
-output neuron backwards through the layers to the input layer in a single
-backpropagation-like pass. A classic example of this is calculating the gradients
+Backpropagation-based methods, in which the signal from a target output neuron is propagated backwards to the input layer, are another way to interpret deep networks that sidestep inefficiencies of the perturbastion-basd methods.
```
Comment (cgreene) [inline sections/06_discussion.md]: I think the references (or nearly all of them) were preserved but the section was dramatically shortened.

File: sections/06_discussion.md [hunk 7]
```diff
@@ -123,93 +119,41 @@ quantifying the change in the output. Umarov et al.
...
 the input and assigns importance scores accordingly. For analyzing images,
 Zeiler and Fergus [@tag:Zeiler2013_visualizing] applied constant-value masks to
-different input patches and studied the changes in the activations of later
-layers. As an alternative to using masks, which can look artificial compared to
-typical values for an input patch, Zintgraf et al.
-[@tag:Zintgraf2017_visualizing] proposed a novel strategy based on marginalizing
-over plausible values of an input patch to more accurately estimate its
-contribution.
+different input patches. More recently, marginalizing over the plausible values
+of an input has been suggest as a way to more accurately estimate contributions
```
Comment (cgreene) [inline sections/06_discussion.md]: 👍

File: sections/06_discussion.md [hunk 8]
```diff
@@ -123,93 +119,41 @@ quantifying the change in the output. Umarov et al.
...
 features that have saturated their contribution to the output, as can happen
 when multiple redundant features are present.
-
 To reduce the computational overhead of perturbation-based approaches, Fong and
 Vedaldi [@tag:Fong2017_perturb] solve an optimization problem using gradient
 descent to discover a minimal subset of inputs to perturb in order to decrease
-the predicted probability of a selected class. When tested on image data, their
-method took about 300 iterations to converge, compared to the ~5000 iterations
-used by LIME. One drawback of this approach is that the use of gradient descent
+the predicted probability of a selected class. Their method converged in many
```
Comment (cgreene) [inline sections/06_discussion.md]: 👍

File: sections/06_discussion.md [hunk 9]
```diff
@@ -123,93 +119,41 @@ quantifying the change in the output. Umarov et al.
...
-architecture, but it is more computationally efficient than integrated
-gradients. Lundberg and Lee [@tag:Lundberg2016_an] noted that several importance
-scoring methods, including DeepLIFT, integrated gradients and LIME, could all be
-considered approximations to the Shapely values [@tag:Shapely], which have a long history in
+@tag:Kindermans2016_investigating]. Networks with Rectified Linear Units (ReLUs) create nonlinearities that must be addressed. Several variants exist for handling this [@tag:Zeiler2013_visualizing @tag:Springenberg2014_striving]. Backpropagation-based methods are a highly active
+area of research. Researchers are still actively identifying
+weaknesses [@tag:Mahendran2016_salient], and new methods are being developed
+to address them [@tag:Selvaraju2016_grad @tag:Sundararajan2017_axiomatic @tag:Shrikumar2017_learning].
+Lundberg and Lee [@tag:Lundberg2016_an] noted that several importance
+scoring methods, integrated gradients and LIME, could all be
```
Comment (cgreene) [inline sections/06_discussion.md]: 👍

File: sections/06_discussion.md [hunk 10]
```diff
@@ -462,34 +400,35 @@ efficiency have come from computation with graphics processing units (GPUs)
 operations so central to deep learning. The massively parallel nature of GPUs
 allows additional optimizations, such as accelerated mini-batch gradient descent
 [@tag:Vanhoucke2011_cpu @tag:Seide2014_parallel @tag:Su2015_gpu
-@tag:Li2014_minibatch]. However, GPUs also have a limited quantity of memory,
-making it difficult to implement networks of useful size and complexity on a
+@tag:Li2014_minibatch]. However, GPUs also have limited memory,
+making networks of useful size and complexity it difficult to implement on a
```
Comment (cgreene) [inline sections/06_discussion.md]: 👍

File: sections/06_discussion.md (hunk 1 above)
Comment (cgreene) [inline sections/06_discussion.md]: 👍

File: sections/06_discussion.md (hunk 2 above)
Comment (cgreene) [inline sections/06_discussion.md]: 👍

File: sections/07_conclusions.md (hunk 3 above)
Comment (cgreene) [inline sections/07_conclusions.md]: made a little change that goes more to what you had

File: sections/07_conclusions.md (hunk 4 above)
Comment (cgreene) [inline sections/07_conclusions.md]: reprhased

**Comment by cgreene** [comment]
plane should board soon - if this looks good feel free to merge or i can merge later

**Comment by cgreene** [comment]
Actually - noticed your point. Will merge & create issue.

**Pull request merged**